content/docs/09-registries-and-packs/7-advanced-configuration.md

```diff
@@ -0,0 +1,211 @@
+embedded keys and values, such as:
+
+```
+version: 0.1
+log:
+  level: info
+  fields:
+    environment: test
+```
+
+
+The key `version` has number value. The `log` key has value with multiple keys (`level`,
+`fields`), which in turn have more keys.
+
+To override the value of `log.level` you can specify an environment variable named
+`REGISTRY_LOG_LEVEL`.
+
+## The default configuration
+
+The docker image for the registy contains the following configuration:
+
+```yaml
+version: 0.1
+log:
+  fields:
+    service: registry
+storage:
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: /data/.spectro-server
+http:
+  addr: :5000
+  headers:
+    X-Content-Type-Options: [ nosniff ]
+    Strict-Transport-Security: [ max-age=63072000; includeSubdomains; preload ]
+    Content-Security-Policy: [ img-src 'self'; script-src 'self';  style-src 'self ]
+    X-Frame-Options: [ DENY ]
+    X-XSS-Protection: [ 1; mode=block ]
+    Referrer-Policy: [ same-origin ]
+auth:
+  htpasswd:
+    realm: basic-realm
+    path: /auth/htpasswd-basic
+```
+
+The server is started with the command `registy serve /etc/spectro/config.yml`.
+You can override this configuration by overriding specific values via environment
+variables or you can pass your own configuration file.
+
+For example you can run the docker container image with the following environment
+variables to override the basic auth realm and logging level:
+
+```bash
+docker run -d \
+    -p 443:5000 \
+    ...
+    -e REGISTRY_LOG_LEVEL=debug \
+    -e REGISTRY_AUTH=htpasswd \
+    -e REGISTRY_AUTH_HTPASSWD_REALM="My Enterprise Realm" \
+    ...
+    gcr.io/spectro-images-public/release/spectro-registry:3.2.0
+```
+
+Alternatively, you can run the image mounting a directory with a new configuration
+file and pointing the server command to it:
+
+```
+docker run -d \
+    -p 443:5000 \
+    ...
+    -v $(pwd)/myconfig.yml:/etc/myconfig.yml
+    ...
+    gcr.io/spectro-images-public/release/spectro-registry:3.2.0
+    /registry serve /etc/myconfig.yml
+```
+## Storage Backend
+
+The pax registry can store data on a file system through a mounted
+volume or it can use S3 storage (AWS or similar).
+
+The following shows a file system backend:
+
+```
+storage:
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: /tmp/registry//data/.spectro-server
+```
+
+S3 Storage needs to know how to connect to S3:
+
+```
+storage:
+  cache:
+    blobdescriptor: inmemory
+  s3:
+    region: us-east-1
+    bucket: my-bucket
+    rootdirectory: /registry
+    encrypt: true|false
+    secure: false|true
+    accesskey: SAMPLEACCESSKEY
+    secretkey: SUPERSECRET
+    host: OPTIONAL_MINIO_HOST_IF_USING
+    port: OPTIONAL_MINIO_PORT_IF_USING
+```
+
+Ephemeral storage for testing purposes is also available:
+```
+storage: inmemory
+```
+
+## Authentication
+
+Basic HTTP Auth is supported:
+
+```
+auth:
+  htpasswd:
+    realm: basic-realm
+    path: /auth/htpasswd-basic
+```
+
+## HTTP
```

Comment:
```suggestion
## Network
```
I wonder if network is a better topic name.

Reply:
I tried keeping the sections with the same name as in the YAML configuration file.

Reply:
We can change to Network but, that title covers address and binding to ports too. Hence, I feel HTTP is more accurate.

Reply:
I follow now. In that case, let's keep it as is.